### PR TITLE
Add board file for Pimoroni Pico Plus 2 W

### LINKED ISSUE #2091: No support for Pimoroni Pico Plus 2 W
[This board](https://shop.pimoroni.com/products/pimoroni-pico-plus-2-w?variant=42182811942995) combines an RP2350B with the same wireless chip in Pico W.

There is not yet a board definition for it. Based on the [schematics](https://cdn.shopify.com/s/files/1/0174/1800/files/Pimoroni_Pico_Plus_2_W_Schematic.pdf?v=1727350279) we can know that they wire an RM2 (Infineon CYW43439) using the same pins used on Pico W and the LED is routed through the chip as well.

### PR DESCRIPTION
**Pull request by gonzalolarralde**
See the board details here: [Pimoroni Pico Plus 2 W](https://shop.pimoroni.com/products/pimoroni-pico-plus-2-w?variant=42182811942995)

This board is similar to pimoroni_pico_plus2_rp2350. I cannot reuse the same header file as this one does not have a direct wire to the LED, similar to pico_w it depends on the WiFi chip for it (compatibility, presumably).

I have no affiliation with Pimoroni, I just purchased a board and I'm using it with this modifications. I see that @ZodiusInfuser has posted some PRs for Pimoroni boards in the past so maybe they can chime in for review?

Fixes #2091

Thanks!

### CODE REVIEW

File: src/boards/include/boards/pimoroni_pico_plus2_w_rp2350.h [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+// --- BOARD SPECIFIC ---
+#define SPICE_SPI 0
+#define SPICE_TX_MISO_PIN 32
+#define SPICE_RX_CS_PIN 33
+#define SPICE_NETLIGHT_SCK_PIN 34
+#define SPICE_RESET_MOSI_PIN 35
+#define SPICE_PWRKEY_BL_PIN 36
+
+#define PIMORONI_PICO_PLUS2_USER_SW_PIN 45
```
Comment (kilograham) [inline src/boards/include/boards/pimoroni_pico_plus2_w_rp2350.h:31]: any reason these aren't PLUS2_W? not sure who uses them!

**Comment by peterharperuk** [comment]
@Gadgetoid

File: src/boards/include/boards/pimoroni_pico_plus2_w_rp2350.h [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// pico_cmake_set PICO_CYW43_SUPPORTED = 1
+
+#ifndef _BOARDS_PIMORONI_PICO_PLUS2_W_RP2350_H
+#define _BOARDS_PIMORONI_PICO_PLUS2_W_RP2350_H
+
+// For board detection
+#define PIMORONI_PICO_PLUS2_W_RP2350
+
+// --- BOARD SPECIFIC ---
+#define SPICE_SPI 0
```
Comment (kilograham) [inline src/boards/include/boards/pimoroni_pico_plus2_w_rp2350.h:24]: Note these are now SPCE_ in other pimoroni board headers

Comment (lurch) [reply src/boards/include/boards/pimoroni_pico_plus2_w_rp2350.h:24]: (see #2048 )

**Comment by lurch** [comment]
This is missing a `// pico_cmake_set_default PICO_RP2350_A2_SUPPORTED` line (which https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pimoroni_pico_plus2_rp2350.h **does** have).

Seems unlikely that `PICO_VBUS_PIN` and `CYW43_PIN_WL_HOST_WAKE` are both on pin 24?

Comparing this to https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pico2_w.h , this seem to be missing `CYW43_DEFAULT_PIN_WL_DATA_OUT`, `CYW43_DEFAULT_PIN_WL_DATA_IN`, etc.?

**Review by kilograham — APPROVED** [review]
Going to merge this as is for 2.1.0 release; it's likely better than nothing

**Comment by kilograham** [comment]
> This is missing a `// pico_cmake_set_default PICO_RP2350_A2_SUPPORTED` line (which https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pimoroni_pico_plus2_rp2350.h **does** have).
> 
> Seems unlikely that `PICO_VBUS_PIN` and `CYW43_PIN_WL_HOST_WAKE` are both on pin 24?
> 
> Comparing this to https://github.com/raspberrypi/pico-sdk/blob/develop/src/boards/include/boards/pico2_w.h , this seem to be missing `CYW43_DEFAULT_PIN_WL_DATA_OUT`, `CYW43_DEFAULT_PIN_WL_DATA_IN`, etc.?

fixed all of these

**Comment by gonzalolarralde** [comment]
Thanks, I initially did this off `picoroni_pico_plus2_rp2350.h` on main, that's why a few things are off. I was completing the changes but seems like @kilograham took a pass on them.

**Comment by kilograham** [comment]
Ah yes, please review my latest change, as I'm about to merge to catch 2.1.0 release

**Comment by kilograham** [comment]
ah, then can you remove them and fix up the SPI pins to be correct

**Comment by kilograham** [comment]
You say "default SPI to the same as pico2_w" ... no reason to assume that they are the same as Pico2 W - are there any SPI pins exposed on the Pico Plus 2 W? I don't have the schematic

**Comment by kilograham** [comment]
actually i looked on the website and this seems reasonable; i'm gonna merge

**Comment by gonzalolarralde** [comment]
@kilograham great, thanks for pushing this through!

**Comment by Gadgetoid** [comment]
Thanks for jumping on this, and nice timing.

I got a bit too caught up getting the wireless stuff working in MicroPython and trying to get a single build for our W and "non-W" variants. (Since both support wireless in theory)

**Pull request merged**